search/django/weedid/tasks.py

```diff
@@ -8,22 +8,31 @@
 from weedid.models import Dataset
 from weedid.utils import make_upload_entity_fields
 from weedid.notification import upload_notification
-from core.settings import THUMBNAILS_DIR, REPOSITORY_DIR, DOWNLOAD_DIR
+from core.settings import THUMBNAILS_DIR, REPOSITORY_DIR, DOWNLOAD_DIR, UPLOAD_DIR
 from pathlib import Path
+import os
+from shutil import move, rmtree
 
 
 @shared_task
-def submit_upload_task(weedcoco_path, image_dir, upload_id):
+def submit_upload_task(weedcoco_path, image_dir, upload_id, new_upload=True):
     upload_entity = Dataset.objects.get(upload_id=upload_id)
-    upload_entity.status = "P"
-    upload_entity.status_details = ""
-
-    # Update fields in database
-    # XXX: maybe this should be delayed
-    with open(weedcoco_path) as f:
-        weedcoco = json.load(f)
-    for k, v in make_upload_entity_fields(weedcoco).items():
-        setattr(upload_entity, k, v)
+    if new_upload:
+        upload_entity.status = "P"
+        upload_entity.status_details = ""
+        # Update fields in database
+        # XXX: maybe this should be delayed
+        with open(weedcoco_path) as f:
+            weedcoco = json.load(f)
+        for k, v in make_upload_entity_fields(weedcoco).items():
+            setattr(upload_entity, k, v)
+    else:
+        with open(weedcoco_path) as f:
+            weedcoco = json.load(f)
+        weedcoco["info"]["metadata"] = upload_entity.metadata
```

Comment:
Ideally we do other modifications here, like updating `info['description']`, but I guess this can be left for now.